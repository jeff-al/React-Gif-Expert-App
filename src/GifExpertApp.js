import { useState } from "react";
import AddCategory from "./components/AddCategory";
import GifGrid from "./components/GifGrid";

const GiftExpertApp = () => {

    const [categories, setCategories] = useState(["Keylor"]);

    return (
        <>
            <h2>Gif Expert App</h2>
            <AddCategory setCategories={ setCategories }/>
            <hr />

            <ol>
                {
                    categories.map((category) => 
                        <GifGrid 
                            category={ category }
                            key={ category }
                        />
                    )
                }
            </ol>
        </>
    );

} 

export default GiftExpertApp;