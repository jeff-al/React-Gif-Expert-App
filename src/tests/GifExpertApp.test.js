import { shallow } from "enzyme";
import GifExpertApp from "../GifExpertApp";

describe('Test for GifExpertApp', () => {
    
    let wrapper = shallow(<GifExpertApp />);

    beforeEach( () => {
        wrapper = shallow(<GifExpertApp />);
    })

    test('should match snapshot', () => {
        
        expect( wrapper ).toMatchSnapshot();
        
    });

    test('should show a list of categories', () => {
        
        const categories = ["Keylor", "Neuer"];
        wrapper = shallow(<GifExpertApp defaultCAtegories={ categories }/>);

        expect( wrapper ).toMatchSnapshot();
        expect( wrapper.find("GifGrid").length ).toBe( categories.length );

    })
    
    

})
