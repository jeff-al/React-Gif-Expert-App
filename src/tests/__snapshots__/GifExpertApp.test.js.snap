// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test for GifExpertApp should match snapshot 1`] = `
<Fragment>
  <h2>
    Gif Expert App
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol />
</Fragment>
`;

exports[`Test for GifExpertApp should show a list of categories 1`] = `
<Fragment>
  <h2>
    Gif Expert App
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol>
    <GifGrid
      category="Keylor"
      key="Keylor"
    />
    <GifGrid
      category="Neuer"
      key="Neuer"
    />
  </ol>
</Fragment>
`;
